bootstrap_storage.go

```
@@ -13,43 +12,31 @@ type BootstrapStorage struct {
 	Reader       io.Reader
 }
 
-func (bs *BootstrapStorage) Load() error {
-	if len(bs.backingStore.data) > 0 || bs.Reader == nil {
-		return nil
+func (ds *BootstrapStorage) Init(backupPath, appName string) {
+	ds.backingStore.Init(backupPath, appName)
+}
+
+func (bs *BootstrapStorage) Load() (*api.FeatureResponse, error) {
```

The logic here needs to change because we now only access that data store when storing or retrieving data. So now we no longer early exit when failing to read the bootstrap, instead we attempt to fallback to the store and if that fails only then do we return an error. In practice I'm not sure the error handling matters much here since we silently ignore the error anyway